File: AndroidStudioProjects/CheckBox/app/src/main/java/com/example/checkbox/MainActivity.java
package com.example.checkbox;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private Button b;
    private CheckBox c1,c2,c3,c4;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        b = findViewById(R.id.button1);

        c1 = findViewById(R.id.checkBox2);
        c2 = findViewById(R.id.checkBox3);
        c3 = findViewById(R.id.checkBox4);
        c4 = findViewById(R.id.checkBox5);

        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                StringBuilder strBuilder = new StringBuilder();
                strBuilder.append("Languages Known : ");
                strBuilder.append("\n");

                if(c1.isChecked()){
                    strBuilder.append(c1.getText().toString());
                    strBuilder.append("\n");
                }
                if(c2.isChecked()){
                    strBuilder.append(c2.getText().toString());
                    strBuilder.append("\n");
                }
                if(c3.isChecked()){
                    strBuilder.append(c3.getText().toString());
                    strBuilder.append("\n");
                }
                if(c4.isChecked()){
                    strBuilder.append(c4.getText().toString());
                    strBuilder.append("\n");
                }

                if(!c1.isChecked() && !c2.isChecked() && !c3.isChecked() && !c4.isChecked()){
                    strBuilder.append("None");
                }

                Toast.makeText(MainActivity.this,strBuilder.toString(),Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: AndroidStudioProjects/ImplicitIntent/app/src/main/java/com/example/implicitintent/MainActivity.kt
package com.example.implicitintent

import android.content.Intent
import android.net.Uri
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.ContactsContract
import android.widget.Button

class MainActivity : AppCompatActivity() {

    private lateinit var browser:Button
    private lateinit var dial:Button
    private lateinit var email:Button
    private lateinit var maps:Button
    private lateinit var contacts:Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initFunction()
        handleEvents()
    }
    private fun initFunction(){
        browser = findViewById(R.id.button_Browser)
        dial = findViewById(R.id.button_Dail)
        email = findViewById(R.id.button_Email)
        maps = findViewById(R.id.button_Maps)
        contacts = findViewById(R.id.button_Contacts)

    }

    private fun handleEvents(){
        browser.setOnClickListener {
            openWebPage("https://kotlinlang.org/docs/map-operations.html#add-and-update-entries")
        }
        dial.setOnClickListener {
            dialPhoneNumber("9876543210")
        }
        email.setOnClickListener {
            composeEmail(arrayOf("a@a.gmail.com"),"Test Mail",Uri.parse("hello.pdf"))
        }
        maps.setOnClickListener {
            showMap(Uri.parse("geo:47.6,-122.3?z=11"))
        }
        contacts.setOnClickListener {
            insertContact("Loki","loki@gmail.com")
        }
    }

    fun openWebPage(url: String) {
        val webpage: Uri = Uri.parse(url)
        val intent = Intent(Intent.ACTION_VIEW, webpage)
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        }
    }

    fun dialPhoneNumber(phoneNumber: String) {
        val intent = Intent(Intent.ACTION_DIAL).apply {
            data = Uri.parse("tel:$phoneNumber")
        }
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        }
    }

    fun composeEmail(addresses: Array<String>, subject: String, attachment: Uri) {
        val intent = Intent(Intent.ACTION_SEND).apply {
            type = "*/*"
            putExtra(Intent.EXTRA_EMAIL, addresses)
            putExtra(Intent.EXTRA_SUBJECT, subject)
            putExtra(Intent.EXTRA_STREAM, attachment)
        }
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        }
    }

    fun showMap(geoLocation: Uri) {
        val intent = Intent(Intent.ACTION_VIEW).apply {
            data = geoLocation
        }
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        }
    }

    fun insertContact(name: String, email: String) {
        val intent = Intent(Intent.ACTION_INSERT).apply {
            type = ContactsContract.Contacts.CONTENT_TYPE
            putExtra(ContactsContract.Intents.Insert.NAME, name)
            putExtra(ContactsContract.Intents.Insert.EMAIL, email)
        }
        if (intent.resolveActivity(packageManager) != null) {
            startActivity(intent)
        }
    }
}